src/NetTopologySuite/IO/GML2/GMLReader.cs

```diff
@@ -177,14 +185,68 @@ protected IEnumerable<Coordinate> ReadPosListAsCoordinates(int numOrdinates, str
             return coordinates;
         }
 
+        protected CoordinateSequence ReadPosList(XmlReader reader, int srsDimension, CoordinateSequenceFactory csFactory)
+        {
+            Assert.IsTrue(IsStartElement(reader, "posList"));
+
+            // Read, update srsDimension
+            srsDimension = ReadSrsDimension(reader, srsDimension);
+            reader.Read();
+
+            var coordinatesSpan = reader.ReadContentAsString().Trim().AsSpan();
```

Comment:
:question: (nit) I haven't pulled this up yet, but does `.AsSpan().Trim()` work with this version of the API?  If so, that's a free small improvement, since it saves an allocation without any downsides.